lib/chef/provider/cron.rb

```diff
@@ -51,6 +67,9 @@ def load_current_resource
         @current_resource = Chef::Resource::Cron.new(@new_resource.name)
         @current_resource.user(@new_resource.user)
         @cron_exists = false
+        # Raise an execption if Chef can't read or update the crontab.
+        can_modify_cron?
```

Comment:
if you look at the assertions API that is used in all the define_resource_requirements methods across our providers that's all that i'm talking about.  there should be lots of examples, its just an assertion that must be true, with an exception to raise in normal mode if its not, and text for why run if its not.